### PR TITLE
Stop leaking temporary netCDF files.

### PR DESCRIPTION
**Pull request by rhattersley**
Discovered while investigating the biggus 0.9 failures in #1560.

### CODE REVIEW

**Comment by pelson** [comment]
```
The file header at lib/iris/tests/__init__.py is out of date. The last commit was in 2015, but the copyright states it was 2014.

The file header at lib/iris/tests/test_netcdf.py is out of date. The last commit was in 2015, but the copyright states it was 2014.
```

:+1:  other than that.

**Comment by rhattersley** [comment]
> The file header at lib/iris/tests/__init__.py is out of date.

That's [karma](https://github.com/SciTools/iris/pull/1596#issuecomment-85956542) for you. :wink:

**Comment by rhattersley** [comment]
Commit amended.

**Comment by bjlittle** [comment]
Yup :+1: @rhattersley

File: lib/iris/tests/test_netcdf.py [hunk 1]
```diff
@@ -25,7 +25,6 @@
 # importing anything else.
 import iris.tests as tests
 
-import os
```
Comment (bjlittle) [inline lib/iris/tests/test_netcdf.py]: @rhattersley the `os` module is used later on ...

Comment (rhattersley) [inline lib/iris/tests/test_netcdf.py]: :unamused: I could have sworn I checked that ... I remember checking *something*!

**Comment by rhattersley** [comment]
Travis has gone a bit bonkers ... the example test sub-task has finished but Travis hasn't noticed. This PR should be OK to merge now.

**Comment by bjlittle** [comment]
@rhattersley Yup, kinda wondered what Travis was doing ...

I'm assuming that this is an expected failure which will be resolved by the latest Biggus?

```
ERROR: test_deferred_bytes (iris.tests.unit.fileformats.pp.test__create_field_data.Test__create_field_data)

----------------------------------------------------------------------

Traceback (most recent call last):

File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/Iris-1.8.0.dev0-py2.7-linux-x86_64.egg/iris/tests/unit/fileformats/pp/test__create_field_data.py", line 74, in test_deferred_bytes

self.assertEqual(field._data.shape, data_shape)

File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/Biggus-0.9.1-py2.7.egg/biggus/__init__.py", line 1018, in shape

return _sliced_shape(self.concrete.shape, self._keys)

File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/Biggus-0.9.1-py2.7.egg/biggus/__init__.py", line 2583, in _sliced_shape

size = len(range(*key.indices(shape[shape_dim])))

TypeError: '_SentinelObject' object cannot be interpreted as an index
```

**Comment by rhattersley** [comment]
That failure should be fixed by #1596. The tests for that should be all clear within a few minutes...

**Comment by rhattersley** [comment]
> That failure should be fixed by #1596.

I'd be very happy to rebase this PR once #1596 is merged.

**Comment by bjlittle** [comment]
I was just going to ask you that :smile: 

Once Travis is happy with #1596, I'll merge and you can rebase ... hopefully Travis plays nicely ...

**Comment by bjlittle** [comment]
@rhattersley #1596 has landed, go rebase :smile:

**Comment by rhattersley** [comment]
Rebase ... awaiting the man from Del Monte ... :tropical_drink:

**Comment by bjlittle** [comment]
He say "yes" :man: :tangerine:

**Pull request merged**